libs/ui-react/src/lib/Components/Spinner/Spinner.tsx

```diff
@@ -0,0 +1,57 @@
+import { cn } from '@ledgerhq/ldls-utils-shared';
+import { cva } from 'class-variance-authority';
+import { forwardRef } from 'react';
+import { IconSize } from '../../Components/Icon/Icon';
+
+export type SpinnerProps = React.SVGProps<SVGSVGElement> & {
+  size?: IconSize;
+  appearance?: 'base' | 'accent';
+};
+
+const iconVariants = cva('shrink-0 animate-spin', {
+  variants: {
+    appearance: {
+      base: 'text-base',
+      accent: 'text-interactive',
+    },
+  },
+  defaultVariants: {
+    appearance: 'base',
+  },
+});
+
+/**
+ * A basic spinner component for loading states.
+ *
+ * @example
+ * <Spinner />
+ *
+ * @default size 16
+ * @default appearance 'base'
+ */
```

Comment:
A couple of points: 

- I personally use `Interface` for React component prop , curious why you use `type` ? I am seeing @gamegee uses `Type` in his props definition as well . let's talk about this and find unified way of doing this. 

- Let try to always documente props inline within TypeScript type definitions (see code suggestion) that's basically what i am doing in this [PR](https://github.com/LedgerHQ/ldls/pull/290). 

- Same for defaults, better to inline them.     

```suggestion
export interface SpinnerProps extends React.SVGProps<SVGSVGElement> {
  /**
   * The size of the spinner icon in pixels.
   * @default 16
   */
  size?: IconSize;
  /**
   * The visual style of the spinner.
   * @default 'base'
   */
  appearance?: 'base' | 'accent';
}

/**
 * A basic spinner component for loading states.
 *
 * @example
 * <Spinner />
 */
```

Reply:
I heavily use unions so I default to types. Actually my liking of `types` as opposed to `interfaces` came exactly from that TS course you linked, @gamegee. Interfaces are more verbose and feel weirdly non-idiomatic IMO. [TS handbook](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces) states it's a matter of preference.

+1 for a eslint rule. We should add [@typescript-eslint/consistent-type-definitions](https://typescript-eslint.io/rules/consistent-type-definitions/) and configure it based on our decision. Let's discuss in the mob.